src/renderer/features/multisigs/model/multisigs-model.ts

```diff
@@ -0,0 +1,177 @@
+import { createEffect, createEvent, sample, scopeBind } from 'effector';
+import { once } from 'patronum';
+import { GraphQLClient } from 'graphql-request';
+
+import {
+  Account,
+  Chain,
+  ChainId,
+  Connection,
+  MultisigAccount,
+  NotificationType,
+  AccountType,
+  ChainType,
+  CryptoType,
+  ExternalType,
+  SigningType,
+  WalletType,
+} from '@shared/core';
+import { networkModel, networkUtils } from '@entities/network';
+import { accountUtils, walletModel, walletUtils } from '@entities/wallet';
+import { MultisigResult, multisigService } from '@entities/multisig';
+import { multisigUtils } from '../lib/mulitisigs-utils';
+import { isEthereumAccountId, toAddress } from '@shared/lib/utils';
+import { CreateParams } from '@entities/wallet/model/wallet-model';
+import { notificationModel } from '@entities/notification';
+
+const multisigsDiscoveryStarted = createEvent();
+const chainConnected = createEvent<ChainId>();
+
+type GetMultisigsParams = {
+  chain: Chain;
+  accounts: Account[];
+};
+
+type StartChainsProps = {
+  chains: Chain[];
+  connections: Record<ChainId, Connection>;
+};
+
+type GetMultisigsResult = {
+  chain: Chain;
+  indexedMultisigs: MultisigResult[];
+};
+
+const connectChainsFx = createEffect(({ chains, connections }: StartChainsProps) => {
+  const boundConnected = scopeBind(chainConnected, { safe: true });
+
+  chains.forEach((chain) => {
+    if (networkUtils.isDisabledConnection(connections[chain.chainId])) return;
+    boundConnected(chain.chainId);
+  });
+});
+
+const getMultisigsFx = createEffect(async ({ chain, accounts }: GetMultisigsParams): Promise<GetMultisigsResult> => {
+  const multisigIndexerUrl =
+    networkUtils.isMultisigSupported(chain.options) && chain.externalApi?.[ExternalType.MULTISIG]?.[0]?.url;
+
+  if (multisigIndexerUrl && accounts.length) {
+    const client = new GraphQLClient(multisigIndexerUrl);
+
+    const indexedMultisigs = await multisigService.filterMultisigsAccountIds(
+      client,
+      accounts.map((account) => account.accountId),
+    );
+
+    return {
+      indexedMultisigs,
+      chain,
+    };
+  }
+
+  return {
+    indexedMultisigs: [],
+    chain,
+  };
+});
+
+const saveMultisigFx = createEffect((multisigsToAdd: CreateParams<MultisigAccount>[]) => {
+  multisigsToAdd.forEach((multisig) => {
+    walletModel.events.multisigCreated(multisig);
+
+    notificationModel.events.notificationsAdded([
+      {
+        read: false,
+        type: NotificationType.MULTISIG_CREATED,
+        dateCreated: Date.now(),
+      },
+    ]);
+  });
+});
+
+sample({
+  clock: [multisigsDiscoveryStarted, once(networkModel.$connections)],
```

Comment:
Why not trigger `multisigsDiscoveryStarted` with some specific interval?
Include `networkModel.$connections` in `source` and filter them to be not empty